xdsl/dialects/onnx.py

```diff
@@ -124,17 +124,15 @@ def __init__(self, operand: SSAValue):
         )
 
     def verify_(self) -> None:
-        if not isinstance(
-            operand_type := self.operand.type, TensorType
-        ) or not isinstance(res_type := self.res.type, TensorType):
-            assert (
-                False
-            ), "onnx elementwise operation operand and result must be of type TensorType"
-            operand_type = cast(TensorType[Attribute], operand_type)
-            res_type = cast(TensorType[Attribute], res_type)
+        assert isinstance(operand_type := self.operand.type, TensorType)
+        assert isinstance(res_type := self.res.type, TensorType)
```

Comment:
Is this really necessary, given the way `operand_type` and `res_type` are defined?

Reply:
you mean the type constraint? We have a few experimental branches that makes this unnecessary but currently SSAValue doesn't have the information about the type of the values it can take at runtime in the type system, so we have to check them every time. We can assert here because the built-in verifier by irdl is called before the custom `verify_` method